### PR TITLE
Share to Twitter

### PR DESCRIPTION
**Pull request by maxkfranz**
Add shareable summary support for Twitter:

- Twitter card support has been added.  To support this feature, dynamic meta tags must be available on the initial HTTP response for the `/document/:id` page.  These meta tags are generated by server-side EJS templates on index.html.
- One of the Twitter card meta tags is a reference to the image for the card.  This is supported by server-side image generation using Cytosnap at the `/api/document/:id.png` route.  The use of Cytosnap required pulling out the base Cytoscape stylesheet as a JSON-serialisable object.
- The readme has been updated to include share-related environment variables.  There are some unrelated variables that are undocumented.  They should be documented as a part of #575.
- Improvements to the read-only view will be part of a different PR in conjunction with data changes related to the admin panel #524.
- The "Share" button in the editor is now more salient.  This should help encourage users to click the button when they're finished editing their documents.  It also matches the Twitter style of buttons to be more familiar to users.

Other, related changes:

- Update Cytoscape to 3.12.0 (latest).
- The Cytoscape stylesheet code is accessible from the client or the server under `util`.
- Environment variables are handled more consistently between the server and the client now.  All variables available to the client are defined in `src/client-env-vars.json`.  You can `import` the variables from `config.js` on the client now.  Before they would not be dynamic unless you used a `process.env.VAR_NAME` string to refer to `VAR_NAME` in the client-side source code.  There is a preliminary list of variables in that JSON file, but we'll have to test and reevaluate.

Ref : Shareable summary #542

Preview:

Screen Shot 2019-11-12 at 4 20 08 PM

Screen Shot 2019-11-12 at 4 20 14 PM

Screen Shot 2019-11-12 at 4 21 36 PM

Screen Shot 2019-11-12 at 4 22 17 PM

### CODE REVIEW

**Comment by maxkfranz** [comment]
@jvwong, this will be difficult to test locally.  Unless you notice any major issues with the code, we should just merge this in to test on the dev CI server.

**Comment by maxkfranz** [comment]
I've set up the environment variables for unstable.factoid.baderlab.org in anticipation of the merge.

It's set up with a "Biofactoid Test App" on Twitter under the [TestAcc87711534 account](https://twitter.com/TestAcc87711534).

N.B.:

- The unstable.factoid.baderlab.org instance has been set up with environment variables to support Twitter in anticipation of merging this PR.  It uses a "Biofactoid Test" app registered with Twitter to post to the [@TestAcc87711534 account](https://twitter.com/TestAcc87711534).  A prod instance will use the @biofactoid account and a "Biofactoid Production" app registration.
- The card UI won't show up in a tweet until the unstable.factoid.baderlab.org is submitted to Twitter.  I can submit it once this PR is merged in.

**Comment by jvwong** [comment]
OK I'll take a peek and merge in ASAP. Will need to rebase my progress in #582 off it...

File: src/model/document/document.js [hunk 1]
```diff
@@ -218,6 +230,10 @@ class Document {
     return this.rwMeta('text', newText);
   }
 
+  reference(){
```
Comment (jvwong) [inline src/model/document/document.js]: I intend to call this 'citation' which returns UI-friendly strings like `{ title, authors, reference }`

**Comment by maxkfranz** [comment]
Here's how it looks on Twitter:

Screen Shot 2019-11-14 at 3 21 58 PM

https://twitter.com/TestAcc87711534/status/1195074153499516928

**Comment by maxkfranz** [comment]
And here's the API route for getting the dynamically-generated demo image that Twitter puts in the card:

https://unstable.factoid.baderlab.org/api/document/demo.png

(`demo` can be any document ID.)

**Pull request merged**